backend/clap/src/main/java/zpi/ppea/clap/service/SubmissionService.java

```diff
@@ -0,0 +1,43 @@
+package zpi.ppea.clap.service;
+
+import data_store.*;
+import lombok.RequiredArgsConstructor;
+import net.devh.boot.grpc.client.inject.GrpcClient;
+import org.springframework.stereotype.Service;
+import zpi.ppea.clap.config.ValueConfig;
+import zpi.ppea.clap.dtos.DetailsSubmissionResponseDto;
+import zpi.ppea.clap.dtos.SubmissionDto;
+import zpi.ppea.clap.logic.BusinessLogicService;
+import zpi.ppea.clap.mappers.DetailedSubmissionMapper;
+import zpi.ppea.clap.mappers.SubmissionMapper;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class SubmissionService {
+    @GrpcClient("dataStore")
+    DataStoreGrpc.DataStoreBlockingStub dataStoreBlockingStub;
+
+    private final ValueConfig valueConfig;
+    private final BusinessLogicService businessLogicService;
+
+    public List<SubmissionDto> getSubmissions() {
+        SubmissionsResponse allSubmissionsGrpc = dataStoreBlockingStub.getSubmissions(
+                SubmissionRequest.newBuilder().setAssessorEmail(valueConfig.getFirebaseEmail()).build()
+        );
+
+        return SubmissionMapper.submissionListToDtos(allSubmissionsGrpc.getSubmissionsList());
+    }
+
+    public DetailsSubmissionResponseDto getDetailedSubmission(Integer submissionId) {
+        DetailsSubmissionResponse detailsSubmissionResponse = dataStoreBlockingStub.getSubmissionDetails(
```

Comment:
We're missing authorization - if the assessor has access to the requested submission.

Reply:
and how do you want to check it? Should I once again ask with email for submisssion endpoint? I don't have more options.

Reply:
I don't see any other option either.